sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -33,44 +76,97 @@ export function buildRequestUrl(
   );
 }
 
-function appendQueryParams(url: string, options: RequestParameters = {}) {
+function getQueryParamValue(
+  key: string,
+  allowReserved: boolean,
+  style: QueryParameterStyle,
+  param: any,
+): string {
+  let separator: string;
+  if (style === "pipeDelimited") {
+    separator = "|";
+  } else if (style === "spaceDelimited") {
+    separator = "%20";
+  } else {
+    separator = ",";
+  }
+
+  let paramValues: any[];
+  if (Array.isArray(param)) {
+    paramValues = param;
+  } else if (typeof param === "object" && param.toString === Object.prototype.toString) {
+    // If the parameter is an object without a custom toString implementation (e.g. a Date),
+    // then we should deconstruct the object into an array [key1, value1, key2, value2, ...].
+    paramValues = Object.entries(param).flat();
+  } else {
+    paramValues = [param];
+  }
+
+  const value = paramValues
+    .map((p) => {
+      if (p === null || p === undefined) {
+        return "";
+      }
+
+      if (!p.toString || typeof p.toString !== "function") {
+        throw new Error(`Query parameters must be able to be represented as string, ${key} can't`);
+      }
+
+      const rawValue = p.toISOString !== undefined ? p.toISOString() : p.toString();
+      return allowReserved ? rawValue : encodeURIComponent(rawValue);
+    })
+    .join(separator);
+
+  return `${allowReserved ? key : encodeURIComponent(key)}=${value}`;
```

Comment:
I feel like I'm missing something here, why do we return the key with no value when `allowReserved` is true?